### ISSUE TITLE
ClassIndex must check for NoClassDefFoundError also

### ISSUE THREAD

**Issue by ptahchiev** [atteo/classindex#56]
At the moment the `ClassIndex` is only checking for a `ClassNotFoundException` during class loading:
https://github.com/atteo/classindex/blob/master/classindex/src/main/java/org/atteo/classindex/ClassIndex.java#L404-L406

However, I believe it should be checking also for `NoClassDefFoundError` (see image):
![noclassdeffound](https://user-images.githubusercontent.com/5320045/68282049-97748b00-0081-11ea-9fc9-cdf6b1c3ea2d.png)

I have a maven project with two modules:
- moduleA
- moduleB defines module A as optional dependency.

Then the client has the option to chose which combination of modules to add in their classpath - moduleA, moduleB, both modules or none of the modules.

If the customer selects moduleB only then I try to find subclasses:
```
ClassFilter.only().topLevel().from(ClassIndex.getSubclasses(AbstractEntityDefinition.class))
```
and then it fails with `NoClassDefFoundError` because the index file in moduleB was created when moduleA was there during compile time but is missing at runtime now.

**Comment by ptahchiev** [comment]
Any idea when the next release will be out?

**Comment by sentinelt** [comment]
I've released version 3.9 just a minute ago. Should appear in Maven Central in about a day.
